modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicStoreFactory.java

```diff
@@ -0,0 +1,116 @@
+
+    private static final String DISPLAY_NAME = "Vector Mosaic Data Store";
+    private static final String DESCRIPTION = "Creates Vector Mosaic from multiple sources";
+    public static final Param REPOSITORY_PARAM =
+            new Param(
+                    "repository",
+                    Repository.class,
+                    "The repository that will provide the store intances",
+                    true,
+                    null,
+                    new KVP(Param.LEVEL, "advanced"));
+    public static final Param NAMESPACE =
+            new Param("namespace", String.class, "Namespace prefix", false);
+
+    public static final Param DELEGATE_STORE_NAME =
+            new Param(
+                    "delegateStoreName",
+                    String.class,
+                    "The name of the delegate store.  "
+                            + "The delegate store must point to vector features with the same attributes/schema.",
+                    true);
+    public static final Param PREFERRED_DATASTORE_SPI =
+            new Param(
+                    "preferredDataStoreSPI",
+                    String.class,
+                    "The preferred DataStoreSPI to test connection parameters against. "
+                            + "If not specified, the first DataStoreSPI found that accepts connection parameters will be used. "
+                            + "Default is org.geotools.data.shapefile.ShapefileDataStoreFactory",
+                    false);
+
+    public static final Param CONNECTION_PARAMETER_KEY =
+            new Param(
+                    "connectionParameterKey",
+                    String.class,
+                    "The key to use for the connection parameter.  Default is 'url'",
+                    false);
+
+    @Override
+    public DataStore createDataStore(Map<String, ?> params) throws IOException {
+        Repository repository = lookup(REPOSITORY_PARAM, params, Repository.class);
+        String namespace = lookup(NAMESPACE, params, String.class);
+        String delegateStoreName = lookup(DELEGATE_STORE_NAME, params, String.class);
+        String preferredDataStoreSPI = lookup(PREFERRED_DATASTORE_SPI, params, String.class);
+        String connectionParameterKey = lookup(CONNECTION_PARAMETER_KEY, params, String.class);
+        VectorMosaicStore vectorMosaicStore =
+                new VectorMosaicStore(
+                        delegateStoreName,
+                        repository,
+                        preferredDataStoreSPI,
+                        connectionParameterKey);
+        vectorMosaicStore.setNamespaceURI(namespace);
+        return vectorMosaicStore;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    @Override
+    public String getDescription() {
+        return DESCRIPTION;
+    }
+
+    @Override
+    public Param[] getParametersInfo() {
+        return new Param[] {REPOSITORY_PARAM, NAMESPACE, DELEGATE_STORE_NAME};
```

Comment:
Does not contain CONNECTION_PARAMETER_KEY, I believe as a result it won't show up in the GeoServer GUI. Did you do some interactive tests in GeoServer?

Reply:
Fixed